smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestRequestGenerator.kt

```diff
@@ -193,39 +175,26 @@ open class HttpProtocolUnitTestRequestGenerator protected constructor(builder: B
     }
 
     private fun renderExpectedHeaders(test: HttpRequestTestCase) {
-        if (test.headers.isNotEmpty()) {
-            writer.openBlock("headers: [")
-                .call {
-                    for ((idx, hdr) in test.headers.entries.withIndex()) {
-                        val suffix = if (idx < test.headers.size - 1) "," else ""
-                        writer.write("\$S: \$S$suffix", hdr.key, hdr.value)
-                    }
-                }
-                .closeBlock("],")
-        }
+        writeHeaders("headers", test.headers)
+        writeHeaders("forbiddenHeaders", test.forbidHeaders)
+        writeHeaders("requiredHeaders", test.requireHeaders)
+    }
 
-        if (test.forbidHeaders.isNotEmpty()) {
-            val forbiddenHeaders = test.forbidHeaders
-            writer.openBlock("forbiddenHeaders: [")
-                .call {
-                    forbiddenHeaders.forEachIndexed { idx, value ->
-                        val suffix = if (idx < forbiddenHeaders.size - 1) "," else ""
-                        writer.write("\$S$suffix", value)
-                    }
-                }
-                .closeBlock("],")
+    private fun writeHeaders(name: String, headers: Map<String, String>) {
```

Comment:
The logic in this method is way simpler than before because it uses `joinToString()` rather than inserting a comma based on element index.